### PR TITLE
Adapt mlpack to require only one include; revamp docs to Markdown, not Doxygen

### PR DESCRIPTION
**Pull request by rcurtin**
After this PR, the following is sufficient to import everything in mlpack:

```
#include <mlpack.hpp>
```

That will be much easier for users.  Of course, this necessitated an overhaul of the README and various tutorials and documentation, which caused a whole host of other changes:

1. No more Doxygen documentation.  Everything is now Markdown, and all links are adapted accordingly.
2. Any references to Doxygen documentation in the code (including in the Markdown bindings) are now updated to refer directly to the source code comments.  (And, since it's not great to send users deep into the source code, that is generally described as a 'last resort' for more documentation.  I hope to improve the documentation further in the future to be a bit more like ensmallen; we're not there quite yet.)
3. The CMake `doc` target and checks for Doxygen are now gone.
4. The `SEE_ALSO("@doxygen/...")` bits are adapted to point to the adapted Markdown tutorials and source in with new `@doc` and `@src` names.
5. A few outdated tutorials are removed entirely.
6. The `MLPACK_ENABLE_ANN_SERIALIZATION` macro is added, and must be set by the user if they want to enable serialization for neural networks.  That support is not there by default, because simply the act of registering each layer with cereal means that we instantiate it, and that can take a huge amount of compile time.  Documentation is updated throughout to point that out.

You can see the new README here: https://github.com/rcurtin/mlpack/tree/one-include

### CODE REVIEW

**Comment by rcurtin** [comment]
(The Doxygen check will fail because there is no Doxygen anymore.  I'll remove that job after this is merged.)

File: src/mlpack/tests/ann/feedforward_network_test.cpp [hunk 1]
```diff
@@ -10,8 +10,8 @@
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
+#define MLPACK_ENABLE_ANN_SERIALIZATION
 #include <mlpack/core.hpp>
-
 #include <mlpack/methods/ann/ann.hpp>
 #include <mlpack/methods/kmeans/kmeans.hpp>
```
Comment (conradsnicta) [inline src/mlpack/tests/ann/feedforward_network_test.cpp:16]: @rcurtin possibly the same issue here -- replace all include statements with `#include <mlpack.hpp>` ?

File: doc/developer/iodoc.md [hunk 2]
```diff
@@ -98,36 +98,36 @@ If those flags are not given during compilation, the following output would be
...
 
-Here is a sample use of those macros, extracted from methods/pca/pca_main.cpp.
+Here is a sample use of those macros, extracted from `methods/pca/pca_main.cpp`.
 (Some details have been omitted from the snippet below.)
 
-@code
+```c++
 #include <mlpack/core.hpp>
 #include <mlpack/core/util/io.hpp>
 #include <mlpack/core/util/mlpack_main.hpp>
```
[the remark this answers is not in the document]
Comment (rcurtin) [reply doc/developer/iodoc.md:133]: Yeah, good idea, done in 00e0dd2.

File: src/mlpack/tests/ann/feedforward_network_test.cpp (hunk 1 above)
Comment (rcurtin) [reply src/mlpack/tests/ann/feedforward_network_test.cpp:16]: I don't mind leaving these as-is; compilation time for the tests will get longer if we use mlpack.hpp everywhere.

**Review by conradsnicta — APPROVED** [review]

**Comment by conradsnicta** [comment]
@rcurtin ok to merge this?  I have a PR in mind that is dependent on this

**Comment by rcurtin** [comment]
Yeah, sorry, I was traveling for a couple days, but it is ready to merge, so I'll do that now.

**Pull request merged**